### PR TITLE
document modelling refactor

### PR DESCRIPTION
**Pull request by logan-markewich**
This is a PR to update llama-hub to use the new Doucment object class from llama-index

This SHOULD NOTE BE MERGED, until the upstream PR is merged [here](https://github.com/jerryjliu/llama_index/pull/6586)

### CODE REVIEW

**Review by jerryjliu — APPROVED** [review]
will hold off on merging until we land the other release

**Comment by logan-markewich** [comment]
hmm, need to retry this PR lol

**Pull request closed without merging**
